content/docs/apis.md

```diff
@@ -2,57 +2,176 @@
+
+## Connecting a Client
+
+This guide assumes you are using Rust and have Rust installed through `rustup` or other means.
+
+To start, open the `Cargo.toml` of your project and add the `tikv-client` as a dependency. You'll also need to depend on `futures`.
+
+<!-- TODO: Use crates.to once published -->
+
+```toml
+[dependencies]
+tikv-client = { git = "https://github.com/tikv/client-rust" }
+futures = "0.1"
+```
+
+In your `src/main.rs` you can then import the raw API as well as the functionality of the `Future` trait so you can utilize it.
+
+In this example we used `raw`, but you can also use `transaction`. The process is the same.
+
+```rust
+use tikv_client::{Config, raw::Client}
+use futures::Future;
+```
+
+After building a `Config`, you can then create a `Client`.
+
+```rust
+let config = Config::new(vec![ // Use more than one PD endpoint!
+    "192.168.0.100:2379",
+    "192.168.0.101:2379",
+    "192.168.0.102:2379",
+]).with_security(              // If using TLS.
+    "root.ca",
+    "internal.cert",
+    "internal.key",
+);
+
+let unconnected_client = Client::new(config);
+```
+
+The value returned at this point is a `Future`. It needs to be resolved before we can directly make calls. If your application is syncronous you can call`.wait()` to block the current task until the future is resolved. If your application is asyncronous you might have better ways of dealing with this.
+
+```rust
+let client = unconnected_client.wait()?; // Block and resolve the future.
+```
+
+With a connected client, you'll be able to make requests from TiKV. This client supports both singlular or batch operations.
+
+You can find the full documentation for the client (and all your dependencies) by running:
+
+<!-- TODO: Link to docs.rs page once published -->
+
+```bash
+cargo doc --package tikv-client --open
+```
 
 ## Raw key-value API {#raw}
 
-The raw key-value API is a simple API that enables you to interact with TiKV
+Using a connected `raw::Client` you can perform actions such as basic `put`, `get`, and `delete`:
 
-To try out the raw key-value API:
+```rust
+// `Key` and `Value` wrap around `Vec<u8>` values.
+// This means data does not need to be UTF-8.
+let key = "TiKV".to_bytes();
+let value = "Astronaut".to_bytes();
 
-1. Install the necessary packages:
+// This creates a future that must be resolved.
+let req = client.put(key, value);
+req.wait()?;
 
-    ```shell
-    go get -v -u github.com/pingcap/tidb/store/tikv
-    ```
+let req = client.get(key);
+let result = req.wait()?;
 
-1. Import the dependency packages:
+// `Value` and `Key` deref to `Vec<u8>`.
+// They should feel natural to work with.
+assert_eq!(result, Some(value));
 
-    ```go
-    import (
-        "fmt"
-        "github.com/pingcap/tidb/config"
-        "github.com/pingcap/tidb/store/tikv"
-    )
-    ```
+let req = client.delete(key);
+req.wait()?;
 
-1. Create a raw key-value client:
+let req = client.get(key).wait()?;
+assert_eq!(result, None);
```

Comment:
Would it be possible to update the client to futures 0.3, and then use `async`/`await` in the examples? It would make them much cleaner.

Reply:
We're not on nightly so we don't get async/await.